File: Fundamentals of C programming/step07/7.1/7.1.06.c
/*

Числа Фибоначчи
Последовательность чисел Фибоначчи определяется следующим образом:
F1=1,F2=1,
F1​=1,F2​=1,
F3=F1+F2,…,Fk=Fk−2+Fk−1.
F3​=F1​+F2​,…,Fk​=Fk−2​+Fk−1​. Посчитать значение NN-го числа Фибоначчи.

Входные данные:
Одно натуральное число NN, (N≤45)(N≤45)

Выходные данные:
Значение NN-го числа Фибоначчи.

Sample Input:

45

Sample Output:

1134903170

*/

#define _CRT_SECURE_NO_WARNINGS                          
#include <stdio.h>

int main()
{
    double N, F1 = 0, F2 = 1, F3 = 0;
    scanf("%lf", &N);
    
    if ((N > 0) && (N <= 2))
    {
        printf("%.lf", F2);
    }
    else
    {
        if ((N>0)&&(N <= 45))
        {
            for (N; N > 1; N = N - 1)
            {
                F3 = F1 + F2;
                F1 = F2;
                F2 = F3;

            } printf("%.lf\n", F3);
        }
    }
    return 0;
}

File: Good, good Python/step 07/7.01/7.1.06.py
'''

Подвиг 8. Напишите функцию, которая проверяет корректность переданного ей email-адреса в виде строки. 
Будем полагать, что адрес верен, если он обязательно содержит символы '@' и '.', 
а все остальные символы могут принимать значения: 'a-z', 'A-Z', '0-9' и '_'. Если email верен, то функция выводит ДА, иначе - НЕТ.

После объявления функции прочитайте (с помощью функции input) строку с email-адресом и вызовите функцию с этим аргументом.

Sample Input:

sc_lib@list.ru

Sample Output:

ДА

'''

import re

def check_email_validity(email):
    pattern = r'^[a-zA-Z0-9_]+@[a-zA-Z0-9]+\.[a-zA-Z]+$'
    if re.match(pattern, email):
        return 'ДА'
    else:
        return 'НЕТ'

email = input()
result = check_email_validity(email)
print(result)
